open_mastr/soap_api/download.py

```diff
@@ -272,38 +275,87 @@ def flatten_dict(data, serialize_with_json=False):
         "Frist": "Wert",
         "WasserrechtAblaufdatum": "Wert",
     }
+    for k, v in flatten_rule_replace.items():
+        if k in dic:
+            dic[k] = dic[k][v]
+
+    return dic
+
+
+def replace_linked_units_with_unit_identifier(dic: dict) -> dict:
+    """If data point in 'dic' has one or more VerknuepfteEinheit or
+    VerknuepfteEinheiten in its respective dict, then take first MaStRNummer
+    and insert it into key VerknuepfteEinheiten (flatten dict).
```

Comment:
This was the explanation I wrote for `dic[k] = dic[k][0][v]`. Since you changed it to you your described functionality of inserting a comma separated string, I would suggest:

```suggestion
    """If data point in 'dic' has one or more VerknuepfteEinheit or
    VerknuepfteEinheiten in its respective dict, the related units (VerknuepfteEinheiten) are inserted as comma-separated strings.
```

Reply:
Thanks for correcting